cli/configssh_test.go

```diff
@@ -123,13 +155,291 @@ func TestConfigSSH(t *testing.T) {
+			},
+		},
+		{
+			name: "When options differ, selecting no preserves previous options",
+			writeConfig: writeConfig{
+				coder: strings.Join([]string{
+					"# This file is managed by coder. DO NOT EDIT.",
+					"#",
+					"# You should not hand-edit this file, all changes will be lost upon workspace",
+					"# creation, deletion or when running \"coder config-ssh\".",
+					"#",
+					"# Last config-ssh options:",
+					"# :ssh-option=ForwardAgent=yes",
+					"#",
+				}, "\n"),
+			},
+			wantConfig: wantConfig{
+				coder: strings.Join([]string{
+					"# This file is managed by coder. DO NOT EDIT.",
+					"#",
+					"# You should not hand-edit this file, all changes will be lost upon workspace",
+					"# creation, deletion or when running \"coder config-ssh\".",
+					"#",
+					"# Last config-ssh options:",
+					"# :ssh-option=ForwardAgent=yes",
+					"#",
+				}, "\n"),
+				coderPartial: true,
+			},
+			matches: []match{
+				{match: "Use new options?", write: "no"},
+				{match: "Continue?", write: "yes"},
+			},
+		},
+		{
+			name: "Do not overwrite unknown coder config",
+			writeConfig: writeConfig{
+				coder: strings.Join([]string{
+					"We're no strangers to love",
+					"You know the rules and so do I (do I)",
+				}, "\n"),
+			},
+			wantConfig: wantConfig{
+				coder: strings.Join([]string{
+					"We're no strangers to love",
+					"You know the rules and so do I (do I)",
+				}, "\n"),
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			var (
+				client    = coderdtest.New(t, &coderdtest.Options{IncludeProvisionerD: true})
+				user      = coderdtest.CreateFirstUser(t, client)
+				version   = coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+				_         = coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+				project   = coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
+				workspace = coderdtest.CreateWorkspace(t, client, user.OrganizationID, project.ID)
+				_         = coderdtest.AwaitWorkspaceBuildJob(t, client, workspace.LatestBuild.ID)
+			)
+
+			// Prepare ssh config files.
+			sshConfigName, coderConfigName := sshConfigFileNames(t)
+			if tt.writeConfig.ssh != "" {
+				sshConfigFileCreate(t, sshConfigName, strings.NewReader(tt.writeConfig.ssh))
+			}
+			if tt.writeConfig.coder != "" {
+				sshConfigFileCreate(t, coderConfigName, strings.NewReader(tt.writeConfig.coder))
+			}
+
+			args := []string{
+				"config-ssh",
+				"--ssh-config-file", sshConfigName,
+				"--test.default-ssh-config-file", sshConfigName,
+				"--test.ssh-coder-config-file", coderConfigName,
+			}
+			args = append(args, tt.args...)
+			cmd, root := clitest.New(t, args...)
+			clitest.SetupConfig(t, client, root)
+
+			pty := ptytest.New(t)
+			cmd.SetIn(pty.Input())
+			cmd.SetOut(pty.Output())
+			done := tGo(t, func() {
+				err := cmd.Execute()
+				if !tt.wantErr {
+					assert.NoError(t, err)
```

Comment:
suggestion, non-blocking: I always find that just asserting an error can lead to fun situations where the behaviour that fails isn't exactly what you expected to fail 😁  I'd suggest also asserting some part of the error is as expected.

Reply:
I do see the point you're making, but I'd prefer to leave as-is. Generally I like to avoid testing error output unless absolutely necessary. Since we're verifying both the error case and the output (unmodified file), and the happy paths in other tests, I think we've got the necessary bases covered here.